### PR TITLE
Refresh deployment-configuration website pages.

### PR DESCRIPTION
**Pull request by ScottTodd**
I started to write a new "compiler tips" page, then I found better places to put those tips within the pages that already exist. While I added the new sections, I proofread the pages and refreshed their structure and style.

(I still plan on adding new pages in this "guides" section)

|  | |
|--------|--------|
| Current section | https://openxla.github.io/iree/guides/deployment-configurations |
| Preview of this PR | https://scotttodd.github.io/iree/guides/deployment-configurations | 

High level changes:

* Add icon prefixes to many sections
* Adjust grammar / phrasing
* Format tables and code blocks
* Remove `--iree-input-type=stablehlo` (auto should work, and TF / StableHLO are not the focus of these pages)
* Add sample `iree-run-module --list_drivers` and `iree-compile --iree-hal-list-target-backends` output
* Call Metal "experimental" and not "stale"
* Add tip for "--iree-llvmcpu-target-cpu-features"

### CODE REVIEW

File: docs/website/docs/guides/deployment-configurations/bare-metal.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ tags:
   - CPU
 ---
 
-# Run on a Bare-Metal Platform
+# Running on a Bare-Metal Platform
 
 IREE supports CPU model execution on bare-metal platforms. That is, platforms
```
Comment (antiagainst) [inline docs/website/docs/guides/deployment-configurations/bare-metal.md]: .. supports model execution via CPU on .. ?

Comment (antiagainst) [inline docs/website/docs/guides/deployment-configurations/bare-metal.md]: s/That is, platforms without/Bare metal platforms have no operating system support/?

(The current wording does form a sentence grammar-wise.)

File: docs/website/docs/guides/deployment-configurations/bare-metal.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Bare-metal deployment typically uses IREE's LLVM compiler target much like the
 [CPU configuration](./cpu.md), but using a limited subset of IREE's CPU HAL
```
Comment (antiagainst) [inline docs/website/docs/guides/deployment-configurations/bare-metal.md]: For L14 (for whatever reason I cannot select it. :( ), ".. uses IREE's LLVM compiler target backends ..."?

File: docs/website/docs/guides/deployment-configurations/bare-metal.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ See [generate.sh](https://github.com/openxla/iree/blob/main/runtime/src/iree/hal
 for example command-line instructions of some common architectures
 
 You can replace the MLIR file with the other MLIR model files, following the
```
Comment (antiagainst) [inline docs/website/docs/guides/deployment-configurations/bare-metal.md:60]: L52 - L62: end with `.` for each sentence?

File: docs/website/docs/guides/deployment-configurations/gpu-vulkan.md [hunk 4]
```diff
@@ -116,80 +90,112 @@ python -m pip install iree-compiler
...
+    to compile towards each GPU architecture. If no triple is specified then a safe
+    but more limited default will be used. We don't support the full spectrum
+    here[^1]; the following table summarizes the
+    currently recognized ones:
 
-GPU Vendor | Target Triple
-:--------: | :-----------:
+| GPU Vendor | Target Triple |
+| ---------- | ------------- |
 ARM Mali GPU | `valhall-g78-android30`
```
Comment (antiagainst) [inline docs/website/docs/guides/deployment-configurations/gpu-vulkan.md]: Add `e.g.,` for both Mali Adreno GPU triples?

File: docs/website/docs/guides/deployment-configurations/index.md [hunk 5]
```diff
@@ -37,13 +37,40 @@ When compiling programs, a list of target backends must be specified via
 | `cuda`         | NVIDIA GPU support via PTX for CUDA | `cuda` |
 | `rocm`         | **Experimental** <br> AMD GPU support via HSACO for ROCm | `rocm` |
 | `webgpu-wgsl`  | **Experimental** <br> GPU support on the Web via WGSL for WebGPU | `webgpu` |
-| `metal-spirv`  | **Stale - see [Issue#4370](https://github.com/openxla/iree/issues/4370)** <br> GPU support on Apple platforms via MSL for Metal | `metal` |
+| `metal-spirv`  | **Experimental** <br> GPU support on Apple platforms via MSL for Metal | `metal` |
```
Comment (antiagainst) [inline docs/website/docs/guides/deployment-configurations/index.md]: `metal` / `metal-spirv`, given both are recognized; also to be consistent with the following list.

File: docs/website/docs/guides/deployment-configurations/index.md [hunk 6]
```diff
@@ -37,13 +37,40 @@ When compiling programs, a list of target backends must be specified via
 | `cuda`         | NVIDIA GPU support via PTX for CUDA | `cuda` |
```
Comment (antiagainst) [inline docs/website/docs/guides/deployment-configurations/index.md]: For L36, `vulkan` / `vulkan-spirv`, given both are recognized; also to be consistent with the following list.

**Review by antiagainst — APPROVED** [review]
Nice! Just some small nits.

**Review by hcindyl — APPROVED** [review]

File: docs/website/docs/guides/deployment-configurations/bare-metal.md (hunk 3 above)
Comment (ScottTodd) [reply docs/website/docs/guides/deployment-configurations/bare-metal.md:60]: Added to the sentences, omitted for the bulleted list items. Looks like general guidance is to add periods for list items, but I've been omitting them. I'll keep the consistency for now.

https://developers.google.com/style/periods#bulleted-lists
https://developers.google.com/style/lists#capitalization-and-end-punctuation

**Pull request merged**